packages/pn-pa-webapp/src/pages/components/NewNotification/PreliminaryInformations.tsx

```diff
@@ -83,128 +83,139 @@ const PreliminaryInformations = ({ notification, onConfirm }: Props) => {
     trackEventByType(TrackEventType.NOTIFICATION_SEND_PAYMENT_MODE, { target: e.target.value });
   };
 
-  useEffect(() => {
+  const fetchGroups = useCallback(() => {
     if (groups.length === 0) {
       void dispatch(getUserGroups(GroupStatus.ACTIVE));
     }
-  }, []);
+  }, [groups]);
```

Comment:
There are cases in which groups can be empty. This condition creates an infinite-loop.
The groups.length condition can be removed and the useCallback dependencies too.

Reply:
OK, I remove both the condition and the dependency.
I usually include in the dependency list all the data items which are read inside the function, since React documentation indicates to do so. In fact, in other projects where I worked before, a warning arises whenever some read datum is not included in a dependency list.
In this particular case, once the condition is removed, the dependency can be removed as well since groups is no longer read. But I still mention that in other points in the pn FE codebase, there are data items which are read in a useCallback / useEffect / useMemo / etc. function, and which are not included in the dependency list.